src/commands/aksStarterWorkflow/deployStarterWorkflowHelper.ts

```diff
@@ -0,0 +1,19 @@
+import path = require("path");
+import * as fs from 'fs';
+import { getExtensionPath } from "../utils/host";
+import * as vscode from 'vscode';
+
+export function configureStarterConfigDataForAKS(
+    resourceName: string,
+    clusterName: string
+): string {
+    const extensionPath = getExtensionPath();
+
+    // Load vscode resoruce yaml file and replace content.
+    const yamlPathOnDisk = vscode.Uri.file(path.join(extensionPath!, 'resources', 'yaml', 'azure-kubernetes-service.yml'));
```

Comment:
In the future it might be better to pull the workflow file from Github directly instead of storing a copy locally.

curl https://api.github.com/repos/tbarnes94/starter-workflows/contents/deployments/azure-kubernetes-service.yml | jq -r ".content" | base64 --decode

Reply:
❤️🙏 Thank you so much for this comment, let me add some background design thoughts Ivan and I had for keeping local copies of files under resources for organisations:

* keeping a local copy under the extension will mean that in case you have no internet then this extension can still generate the files for template action. Also will avoid any file read failure in case other end point is unavailable.
* another advantage will be in case the online version changed and has breaking impact atleast the local copy of extension will keep working.

What do you think about above rationals, feel free to share thoughts please or I can ping you in teams sometime today. thank you for taking a look and really appreciate it your time for this. 🙏

Reply:
I appreciate the background and rationale. 

Being able to work for users with no internet makes lots of sense. A workaround might be favoring the pull method but still having a local copy in case the user is offline. An alternative approach could be to setup a Github action in this repo that would automatically create a PR whenever the upstream starter workflow changed.

This could be done with [this](https://github.com/peter-evans/create-pull-request/blob/main/docs/examples.md#spider-and-download-a-website) and would look like:

```md
name: Update starter workflow
on:
  schedule:
    - cron:  '0 10 * * *'
jobs:
  format:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Download starter workflow
        run: |
          curl https://api.github.com/repos/tbarnes94/starter-workflows/contents/deployments/azure-kubernetes-service.yml | jq -r ".content" | base64 --decode > resources/yaml/azure-kubernetes-service.yml
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
          commit-message: update starter workflow
          title: Automated updates to starter workflow
```

This approach would also ensure that there would be no breaking changes since you'd be able to review any generated PRs. Although, either approach should be fine because in theory there should be no breaking impacts possible. The only breaking impact I can think of is the variable names changing.

Reply:
Cool sounds good, let’s chant more about this and once we have scoped good possible ideas we can scout them in separate PR. Thank you so much for this, closing this for now. ❤️🙏